src/pageEditor/tabs/modVariablesDefinition/ModVariablesDefinitionEditor.test.tsx

```diff
@@ -0,0 +1,157 @@
+import { actions as editorActions } from "@/pageEditor/store/editor/editorSlice";
+import { actions as modComponentsActions } from "@/store/modComponents/modComponentSlice";
+import { screen, waitFor } from "@testing-library/react";
+import { modInstanceFactory } from "@/testUtils/factories/modInstanceFactories";
+import { modDefinitionFactory } from "@/testUtils/factories/modDefinitionFactories";
+import { mapModInstanceToActivatedModComponents } from "@/store/modComponents/modInstanceUtils";
+import AssignModVariable from "@/bricks/effects/assignModVariable";
+import { uuidv4 } from "@/types/helpers";
+import brickRegistry from "@/bricks/registry";
+import userEvent from "@testing-library/user-event";
+
+beforeEach(() => {
+  brickRegistry.register([new AssignModVariable()]);
+});
+
+describe("ModVariablesDefinitionEditor", () => {
+  it("renders no variables", () => {
+    render(<ModVariablesDefinitionEditor />, {
+      setupRedux(dispatch) {
+        const formState = formStateFactory();
+        dispatch(editorActions.addModComponentFormState(formState));
+        dispatch(editorActions.setActiveModId(formState.modMetadata.id));
+      },
+    });
+
+    expect(
+      screen.getByRole("button", { name: "Add new mod variable" }),
+    ).toBeInTheDocument();
+  });
+
+  it("renders dirty variables", async () => {
+    render(<ModVariablesDefinitionEditor />, {
+      setupRedux(dispatch) {
+        const formState = formStateFactory();
+        dispatch(editorActions.addModComponentFormState(formState));
+        dispatch(editorActions.setActiveModId(formState.modMetadata.id));
+        dispatch(
+          editorActions.editModVariablesDefinition({
+            schema: {
+              properties: {
+                declaredAny: {
+                  description: "Test description",
+                },
+              },
+            },
+          }),
+        );
+      },
+    });
+
+    expect(
+      screen.getByRole("cell", { name: "declaredAny" }),
+    ).toBeInTheDocument();
+  });
+
+  it("renders declared variables", async () => {
+    render(<ModVariablesDefinitionEditor />, {
+      setupRedux(dispatch) {
+        const modInstance = modInstanceFactory({
+          definition: modDefinitionFactory({
+            variables: {
+              schema: {
+                properties: {
+                  declaredAny: {
+                    description: "Test description",
+                  },
+                },
+              },
+            },
+          }),
+        });
+
+        const [modComponent] =
+          mapModInstanceToActivatedModComponents(modInstance);
+
+        dispatch(modComponentsActions.UNSAFE_setModComponents([modComponent!]));
+        dispatch(
+          editorActions.setActiveModId(modInstance.definition.metadata.id),
+        );
+      },
+    });
+
+    await waitFor(() => {
+      expect(
+        screen.getByRole("cell", { name: "declaredAny" }),
+      ).toBeInTheDocument();
+    });
+  });
```

Comment:
NIT: Prefer `findBy` > `waitFor` when awaiting an element:

```suggestion
    await expect(
        screen.findByRole("cell", { name: "declaredAny" }),
      ).resolves.toBeInTheDocument();
  });
```